src/components/frame/v5/pages/UserProfilePage/partials/UserAccountForm/hooks.tsx

```diff
@@ -0,0 +1,207 @@
+import { useAppContext } from '~hooks';
+import { useUpdateUserProfileMutation } from '~gql';
+import {
+  UseAvatarUploaderProps,
+  useGetUploaderText,
+} from '~v5/common/AvatarUploader/hooks';
+import { formatText } from '~utils/intl';
+import Toast from '~shared/Extensions/Toast';
+import Avatar from '~v5/shared/Avatar';
+import {
+  profileFileOptions,
+  MAX_BIO_CHARS,
+  MAX_DISPLAYNAME_CHARS,
+} from '../consts';
+import IconSuccessContent from '../IconSuccessContent';
+import { RowGroup } from '../Row/types';
+import { useUserProfile } from '../UserAccountPage/hooks';
+
+export const useUserProfilePageForm = () => {
+  const { user } = useAppContext();
+  const { avatarUrl, canChangeUsername, daysTillUsernameChange } =
+    useUserProfile();
+  const { formState, register, setValue, getValues, reset } = useFormContext();
+  const { updateUser, wallet } = useAppContext();
+  const [updateAvatar] = useUpdateUserProfileMutation();
+  const displayNameDirty = formState?.dirtyFields.displayName;
+  const showNameMessage: boolean =
+    displayNameDirty && !formState?.errors.displayName?.message;
+
+  useEffect(() => {
+    reset({
+      hasDisplayNameChanged: false,
+      bio: user?.profile?.bio || '',
+      displayName: user?.profile?.displayName || '',
+      location: user?.profile?.location || '',
+      website: user?.profile?.website || '',
+    });
+  }, [user, reset]);
+
+  const updateAvatarFn: UseAvatarUploaderProps['updateFn'] = async (
+    avatar,
+    thumbnail,
+    setProgress,
+  ) => {
+    const formData = new FormData();
+    formData.append('file', avatar || '');
+    setProgress(0);
+
+    /* Axios upload currently has no destination */
+    /* axiosUpload(avatar, thumbnail, setProgress); */
+
+    await updateAvatar({
+      variables: {
+        input: {
+          // @ts-ignore
+          id: user?.walletAddress,
+          avatar,
+          thumbnail,
+        },
+      },
+    });
+
+    await updateUser?.(user?.walletAddress, true);
+
+    toast.success(
+      <Toast
+        type="success"
+        title={{ id: 'upload.avatar.successfully.toast.title' }}
+        description={{
+          id: 'upload.avatar.successfully.toast.description',
+        }}
+      />,
+    );
+
+    if (avatar === null) {
+      setProgress(0);
+    } else {
+      /* Needed in order to signal the upload was successful */
+      setProgress(100);
+    }
+  };
+
+  /*
+   * Imperatively update the hasDisplayNameChanged form field, in order to be able to conditionally
+   * validate the displayName field. We don't want it to be validated if it hasn't changed.
+   */
+  const hasDisplayNameChanged = getValues('hasDisplayNameChanged');
+
+  if (displayNameDirty && !hasDisplayNameChanged) {
+    setValue('hasDisplayNameChanged', true, {
+      shouldValidate: false,
+    });
+  } else if (!displayNameDirty && hasDisplayNameChanged) {
+    setValue('hasDisplayNameChanged', false, {
+      shouldValidate: false,
+    });
+  }
```

Comment:
Would it be possible to put the original display name in hook-form state (either as another field or preferably a value in the context provided to yup) so that the validation can derive `hasDisplayNameChanged` itself? This way the validation could be kept all in one place.

Reply:
@arrenv I haven't changed anything in Tab component, just moved styles from module.css file 
@jakubcolony these are not my changes. I just moved these changes to the hook file.

Reply:
@jakubcolony this is not may changes, I just moved this hook to different file. I suggest refactor it in separate pull request.